### PR TITLE
.travis.yml: fix docker deploy

### PR DESCRIPTION
**Pull request by junghans**
See travis-ci/travis-ci#10512 for details.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
           env: DISTRO=ubuntu EXTERNAL=ON BUILD_UG=ON
 
 after_success:
-  - test "${TRAVIS_BUILD_STAGE_NAME}" = "Deploy" || travis_terminate 0
+  - if [[ "${TRAVIS_BUILD_STAGE_NAME}" =! [dD]eploy ]]; then travis_terminate 0; fi
```
Comment (jkrajniak) [inline .travis.yml:50]: Shouldn't be !=

Comment (junghans) [reply .travis.yml:50]: It was 'or' (`||`), now it is 'and' due to the if statement hence I had to negate the condition.

Comment (jkrajniak) [reply .travis.yml:50]: Just the condition `=!` ? shouldn't be `!=`

Comment (junghans) [reply .travis.yml:50]: Ah....good point.

Comment (junghans) [reply .travis.yml]: Fixed

**Pull request merged**